### PR TITLE
feat: adding redis SSL and username authentification

### PR DESCRIPTION
**Pull request by Yoann-tildev**
### SUMMARY
Added compatibility with Redis 6 new authentification (SSL + ACL)

### TESTING INSTRUCTIONS
Deploying Superset with Redis 6 (we use managed instance from OVH Cloud)

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: helm/superset/templates/_helpers.tpl [hunk 1]
```diff
@@ -92,9 +99,12 @@ WTF_CSRF_TIME_LIMIT = 60 * 60 * 24 * 365
 class CeleryConfig(object):
   CELERY_IMPORTS = ('superset.sql_lab', )
   CELERY_ANNOTATIONS = {'tasks.add': {'rate_limit': '10/s'}}
-{{- if .Values.supersetNode.connections.redis_password }}
-  BROKER_URL = f"redis://:{env('REDIS_PASSWORD')}@{env('REDIS_HOST')}:{env('REDIS_PORT')}/0"
-  CELERY_RESULT_BACKEND = f"redis://:{env('REDIS_PASSWORD')}@{env('REDIS_HOST')}:{env('REDIS_PORT')}/0"
+{{- if .Values.supersetNode.connections.redis_ssl }}
+  BROKER_URL = f"rediss://{env('REDIS_USERNAME')}:{env('REDIS_PASSWORD')}@{env('REDIS_HOST')}:{env('REDIS_PORT')}/0"
```
Comment (dpgaspar) [inline helm/superset/templates/_helpers.tpl:103]: Can you add `REDIS_SCHEME` here to make it possible for an admin to choose between `redis` and `rediss`

File: helm/superset/values.yaml [hunk 2]
```diff
@@ -35,7 +35,7 @@ bootstrapScript: |
   rm -rf /var/lib/apt/lists/* && \
   pip install \
     psycopg2-binary==2.9.1 \
-    redis==3.5.3 && \
+    redis==4.3.4 && \
```
Comment (dpgaspar) [inline helm/superset/values.yaml:38]: our current base requirements is: https://github.com/apache/superset/blob/master/requirements/base.txt#L242

File: helm/superset/templates/_helpers.tpl [hunk 3]
```diff
@@ -69,14 +69,21 @@ def env(key, default=None):
 
 MAPBOX_API_KEY = env('MAPBOX_API_KEY', '')
 CACHE_CONFIG = {
-      'CACHE_TYPE': 'redis',
+      'CACHE_TYPE': 'RedisCache',
```
Comment (dpgaspar) [inline helm/superset/templates/_helpers.tpl:72]: we probably should change: https://github.com/apache/superset/blob/04dd8d414db6a3cddcd073ad74acb2a4b7a53b0b/docker/pythonpath_dev/superset_config.py

**Review by dpgaspar — COMMENTED** [review]
Superset's base dependencies is loose by default on `setup.py` we don't pin or restrict Redis there, but our test suit and base requirements are pinned to 3.5.3 this could be a breaking change. I think that helm should be in sync with the rest of the repo.

**Comment by Yoann-tildev** [comment]
> Superset's base dependencies is loose by default on `setup.py` we don't pin or restrict Redis there, but our test suit and base requirements are pinned to 3.5.3 this could be a breaking change. I think that helm should be in sync with the rest of the repo.

3.5.3 is more than 2 years old and doesn't support Redis 6 SSL authentication. Shouldn't it be the time to upgrade this dependency in the test suit ?

File: helm/superset/templates/_helpers.tpl (hunk 1 above)
Comment (Yoann-tildev) [reply helm/superset/templates/_helpers.tpl:103]: I am unsure of what you're asking. I introduced "redis_ssl" and I'm checking if the option is set in order to be able to choose between redis and rediss (ssl mode) already. 

What's REDIS_SCHEME ?

File: helm/superset/templates/_helpers.tpl (hunk 3 above)
Comment (Yoann-tildev) [reply helm/superset/templates/_helpers.tpl:72]: this change was introduced in version 1.10.0 of flask-caching and is mandatory in version 2.X

as I'm using helm I didn't edit the docket-compose setup but he should be safe to do it I guess

**Comment by Yoann-tildev** [comment]
> > Superset's base dependencies is loose by default on `setup.py` we don't pin or restrict Redis there, but our test suit and base requirements are pinned to 3.5.3 this could be a breaking change. I think that helm should be in sync with the rest of the repo.
> 
> 3.5.3 is more than 2 years old and doesn't support Redis 6 SSL authentication. Shouldn't it be the time to upgrade this dependency in the test suit ?

I investigated : neither Flask-caching nor CacheLib are specifying a specific version of py-redis package. Actually Flask-caching wasn't even specifying a version for CacheLib before v2.0 which specify 0.9.0 https://github.com/pallets-eco/flask-caching/blob/v2.0.0/setup.py#L6

File: helm/superset/templates/_helpers.tpl (hunk 1 above)
Comment (dpgaspar) [reply helm/superset/templates/_helpers.tpl:103]: following this convention: https://www.rfc-editor.org/rfc/rfc3986.html#section-3 for scheme
optional but wouldn't be simpler to just create `REDIS_SCHEME`?

**Comment by dpgaspar** [comment]
> > Superset's base dependencies is loose by default on `setup.py` we don't pin or restrict Redis there, but our test suit and base requirements are pinned to 3.5.3 this could be a breaking change. I think that helm should be in sync with the rest of the repo.
> 
> 3.5.3 is more than 2 years old and doesn't support Redis 6 SSL authentication. Shouldn't it be the time to upgrade this dependency in the test suit ?

Totally agree, can you please upgrade the versions on all the places I mentioned?

**Comment by moreiramarti** [comment]
:wave: 
I'm interested about that.
Are there any news ?

**Comment by villebro** [comment]
@Yoann-tildev is this PR still relevant? I see the config properties are based on the old version e.g. `BROKER_URL` vs `broker_url` which is the new contvention etc. Happy to help review and get this merged if this is still needed, otherwise I propose closing this.

**Comment by rusackas** [comment]
Having not heard about the current relevance of the PR, and since it was proposed to close this 10 months ago, I'll just do so now. Happy to reopen if it's warranted.

**Pull request closed without merging**